test/tests/kernels/curlcurl.i

```diff
@@ -67,16 +67,19 @@
     type = MFEMVectorFunctionCoefficient
     function = exact_e_field
   []
-  [VolumetricSourceCoefficient]
-    type = MFEMVectorFunctionCoefficient
-    function = forcing_field
-  []
 []
 
-[Coefficients]
-  [one]
-    type = MFEMConstantCoefficient
-    value = 1.0
+
+[Materials]
+  [Substance]
+    type = MFEMGenericConstantMaterial
+    prop_names = one
+    prop_values = 1.0
+  []
+  [Forcing]
+    type = MFEMGenericFunctionVectorMaterial
+    prop_names = VolumetricSourceCoefficient
+    prop_values = forcing_field
   []
```

Comment:
The forcing field here is not really a material property - effectively it is just a function put into a coefficient so MFEM can use as a a (volumetric) source field.

Is it worth creating an MFEMFunction to handle this instead, or otherwise allow functions to instatiate properties without a material? This is a similar nomenclature issue to the type we will have when passing functions to boundary conditions for (eg.) a Dirichlet BC setting temperature on a surface (which isn't a material, but will still need a FunctionCoefficient).

Reply:
How is this handled within MOOSE itself? For boundary conditions, the MOOSE classes take a function name as a parameter and I was planning to refactor ours to do the same.

Reply:
Similar to boundary conditions, the kernels that use them take a function name (or names) - an example is [VectorBodyForce](https://github.com/idaholab/moose/blob/next/framework/src/kernels/VectorBodyForce.C), which is adding an equivalent integrator to Platypus's `MFEMVectorFEDomainLFKernel`.  

In fact - every time MOOSE uses a function that I can think of, MFEM would use a FunctionCoefficient (or VectorFunctionCoefficient).

Reply:
OK, I've reverted those changes now.